src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -205,6 +205,11 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository, TextE
         }
         TextExercise textExerciseBeforeUpdate = textExerciseRepository.findByIdElseThrow(textExercise.getId());
 
+        // Forbid changes of course exercise belongs to.
+        if (!textExerciseBeforeUpdate.getCourseViaExerciseGroupOrCourseMember().getId().equals(textExercise.getCourseViaExerciseGroupOrCourseMember().getId().longValue())) {
```

Comment:
please use `!Objects.equals(id1, id2)`
see https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-

Reply:
Thank you. Done 👍 @krusche